src/test/java/com/meilisearch/integration/SearchTest.java

```diff
@@ -486,4 +488,21 @@ public void testPlaceHolderWithLimit() throws Exception {
 
         assertEquals(10, searchResult.getHits().size());
     }
+
+    /** Test search matchingStrategy */
+    @Test
+    public void testSearchMatchingStrategy() throws Exception {
+        String indexUid = "SearchMatchingStrategy";
+        Index index = client.index(indexUid);
+
+        TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+        TaskInfo task = index.addDocuments(testData.getRaw());
+
+        index.waitForTask(task.getTaskUid());
+        SearchRequest searchRequest =
+                new SearchRequest("and").setMatchingStrategy(MatchingStrategy.ALL);
+        SearchResult searchResult = index.search(searchRequest);
+
+        assertEquals(20, searchResult.getHits().size());
```

Comment:
I think you could avoid testing by calling meilisearch in this case, so you could just assert the result from `SearchResult`. 

If you just assert `searchResult.toString()` you can check if the searchResult has the correct value of "all".

Reply:
But `searchResult` does not contain `matchingStrategy` how did I check that "the searchResult has the correct value of "all"?

Reply:
I'm sorry, I meant `searchRequest`!